templates/index/_partials/section.html.twig

```diff
@@ -1,33 +1,71 @@
-{% set pad_length = items|length > 3 ? max_results : 3 %}
+<link rel="stylesheet" href="https://unpkg.com/swiper@8/swiper-bundle.min.css"/>
 
-<section class="homepage-restaurants">
+<script type="module">
+
+import Swiper from 'https://unpkg.com/swiper@8/swiper-bundle.esm.browser.min.js';
+
+const swiper = new Swiper('.swiper', {
+  slidesPerView: 1.25,
+  spaceBetween: 2,
+  slidesPerGroup: 1,
+  navigation: {
+    nextEl: '.swiper-nav-next',
+    prevEl: '.swiper-nav-prev',
+  },
+  lazyLoading: true,
+  breakpoints: {
+    480: {
+      slidesPerView: 2.25,
+      spaceBetween: 2,
+      slidesPerGroup: 1,
+    },
+    768: {
+      slidesPerView: 3.25,
+      spaceBetween: 2,
+      slidesPerGroup: 2,
+    },
+    992: {
+      slidesPerView: 4.25,
+      spaceBetween: 3,
+      slidesPerGroup: 2,
+    },
+    1200: {
+      slidesPerView: 5.25,
+      spaceBetween: 4,
+      slidesPerGroup: 3,
+    },
+  },
+  observer: true, // to be initialized properly inside a hidden container
+  observeParents: true,
+  on: {
+    afterInit: function() {
+      // we need to hide the swiper at the begining because until is not initialized the images do not look very well
+      document.querySelectorAll('.homepage-restaurants').forEach(element => element.classList.remove('hidden'))
+    }
+  }
+});
+</script>
```

Comment:
This would be included multiple times. It should be managed with Webpack, and moved to `js/app/restaurant/list.js`